### PR TITLE
feat: Add keycloak parser, scenario and collection

### PR DESCRIPTION
**Pull request by lperdereau**
This PR consists of implementing:

- [x]  A keycloak logs parser `parsers/s01-parse/inherent-io/keycloak-logs`
- [x]  A keycloak scenario for quick bruteforce attack `scenarios/inherent-io/keycloak-bf`
- [x]  A keycloak scenario for slow bruteforce attack `scenarios/inherent-io/keycloak-slow-bf`
- [x]  A keycloak collection `collection/inherent-io/keycloak`

### CODE REVIEW

File: collections/inherent-io/keycloak.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Keycloak support : parser and brute-force detection
```
Comment (LaurenceJJones) [inline collections/inherent-io/keycloak.md:1]: Provide a configuration example? Does it log to a file if so where is the file location?

```
filenames:
  - /path/to/log
labels:
  type: keycloak
  ```
  Is it a docker program
 ```
 source: docker
 container_name:
   - keycloak
 labels:
   type: keycloak
   ```

File: parsers/s01-parse/inherent-io/keycloak-logs.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+filter: 1 == 1
```
Comment (blotus) [inline parsers/s01-parse/inherent-io/keycloak-logs.yaml:1]: Please filter on the actual program name

File: parsers/s01-parse/inherent-io/keycloak-logs.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+debug: true
```
Comment (blotus) [inline parsers/s01-parse/inherent-io/keycloak-logs.yaml:2]: Remove or comment this

File: scenarios/inherent-io/keycloak-bf.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+groupby: evt.Meta.source_ip
+blackhole: 1m
+reprocess: true
+labels:
+ service: keycloak
+ type: bruteforce
+ remediation: true
+---
+type: leaky
+name: inherent-io/keycloak-bf
```
Comment (blotus) [inline scenarios/inherent-io/keycloak-bf.yaml:16]: Should probably be `inherent-io/keycloak-user-enum` ?

File: scenarios/inherent-io/keycloak-slow-bf.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+groupby: evt.Meta.source_ip
+blackhole: 1m
+reprocess: true
+labels:
+ service: keycloak
+ type: bruteforce
+ remediation: true
+---
+type: leaky
+name: inherent-io/keycloak-slow-bf
```
Comment (blotus) [inline scenarios/inherent-io/keycloak-slow-bf.yaml:16]: Same as above

**Review by LaurenceJJones — CHANGES_REQUESTED** [review]
Reviewed, please see requested changes

File: parsers/s01-parse/inherent-io/keycloak-logs.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (lperdereau) [reply parsers/s01-parse/inherent-io/keycloak-logs.yaml:1]: @blotus and you're asking me to change my filter to one based on the program name.
But I haven't understood how crowdsec can find out the name of the program from a file or a stdout container.

[the remark this answers is not in the document]
Comment (lperdereau) [reply parsers/s01-parse/inherent-io/keycloak-logs.yaml:1]: This thing configure the evt.Parsed.program ?

```
labels:
  type: keycloak
```

**Comment by lperdereau** [comment]
@blotus, @LaurenceJJones  Thank you for your review, I've learnt a lot.

**Review by LaurenceJJones — APPROVED** [review]
LGTM, bitwarden test is flakey

**Review by blotus — APPROVED** [review]

**Pull request merged**